internal/cmd/addhooks/prepush.go

```diff
@@ -0,0 +1,23 @@
+package addhooks
+
+var prePushScript = []byte(`#!/bin/sh
+
+remote="$1"
+url="$2"
+
+if ! command -v gittuf > /dev/null
+then
+    echo "gittuf could not be found"
+    echo "Download from: https://github.com/gittuf/gittuf"
+    exit 1
+fi
+
+echo "Updating RSL."
+gittuf rsl remote pull origin
+echo "Creating new RSL record for HEAD."
+gittuf rsl record HEAD
+
+echo "Pushing refs/gittuf/reference-state-log with ${remote}"
+# --no-verify disables pre-push hook, to prevent infinite loop.
+git push --no-verify ${remote} refs/gittuf/reference-state-log
```

Comment:
Interesting, I wonder if `gittuf rsl remote push` bypasses the hook trigger anyway. I'm reasonably certain it should.

Reply:
I tried, and somehow `gittuf rsl remote push` does not discover my push credentials. Maybe a bug we should file? I can do more investigation tomorrow.

Reply:
Very strange, more details would be great for this.

Reply:
```
datosh@lenovo-x1:~/code/testing-gittuf/repo$ git push --no-verify origin refs/gittuf/reference-state-log 
Everything up-to-date
datosh@lenovo-x1:~/code/testing-gittuf/repo$ gittuf rsl remote push origin
Error: unable to push RSL
authentication required
Usage:
  gittuf rsl remote push <remote> [flags]

Flags:
  -h, --help   help for push

Global Flags:
      --verbose   enable verbose logging
```

`gittuf` does not seem to provide a lot of details, even with `--verbose` flag.

As I authenticated to this repository using VSCode, and I had similar issues in the past with other tools, I suspect VSCode is doing something non-standard. I will do some investigation, but don't hesitate to provide any pointers :)

Reply:
The `--verbose` flag doesn't do much yet. We have the logging framework merged as of a week or so ago, but now we need to go in and add Debug statements.

Reply:
gittuf * push has worked with the standard CLI invocation where I'm authenticated using ssh-agent and so. Nothing jumps out at me atm re the VSCode thing.

Reply:
> Interesting, I wonder if `gittuf rsl remote push` bypasses the hook trigger anyway. I'm reasonably certain it should.

Yes, it does. I have changed the hook script to only rely on `gittuf`.

> gittuf * push has worked with the standard CLI invocation where I'm authenticated using ssh-agent and so. Nothing jumps out at me atm re the VSCode thing.

It seems to be related to my particular VSCode setup. I see the same auth issue with other tools.
On the other hand, when I checkout with ssh key everything works fine. I think we can ignore that auth error in this context.